odl/operator/tensor_ops.py

```diff
@@ -337,7 +337,14 @@ def derivative(self, vf):
         inner_vf = vf.copy()
 
         for gi in inner_vf:
-            gi /= vf_pwnorm_fac * gi ** (self.exponent - 2)
+            if self.exponent >= 2:
+                tmp = vf_pwnorm_fac * gi ** (self.exponent - 2)
+                # Any component that is zero is not divided with
+                for i, val in zip(np.arange(len(tmp)), tmp):
+                    if not val == 0.0:
```

Comment:
This is gonna go away anyway, but just a small comment: If there is a comparison operator for the opposite, use it, instead of prepending `not`: `val != other_val` rather than `not val == other_val`. More readable.